graph/topo/johnson_cycles.go

```diff
@@ -107,6 +215,47 @@ func (j *johnson) circuit(v int) bool {
 	return f
 }
 
+// circuitMaxLen is the CIRCUIT sub-procedure in the paper but produces circuits up to a maximum length.
+func (j *johnson) circuitMaxLen(v int, maxLen int) bool {
+	f := false
+	// TODO: can we replace the "+ 2" with "+ 3" or "+ 4"?
```

Comment:
Can you clarify this? It wouldn't be enough information for me to come back to in the future.

Reply:
It can be ignored. The TODO was a note to myself to think about replacing `len(j.stack)+2` with `len(j.stack)+3`. But I have realized that the latter would not work. I will delete this comment